central/cluster/datastore/datastore_impl_postgres_test.go

```diff
@@ -68,6 +843,132 @@ func (s *ClusterPostgresDataStoreTestSuite) TestSearchClusterStatus() {
 	s.Equal(0, len(res))
 }
 
+func (s *ClusterPostgresDataStoreTestSuite) TestAddDefaults() {
+
+	s.Run("Error on nil cluster", func() {
+		s.Error(addDefaults(nil))
+	})
+
+	s.Run("Some default values are set for uninitialized fields", func() {
+		cluster := &storage.Cluster{CentralApiEndpoint: centralEndpoint, MainImage: mainImage}
+		s.NoError(addDefaults(cluster))
+		s.Empty(cluster.GetCollectorImage()) // must not be set
+		s.Equal(centralEndpoint, cluster.GetCentralApiEndpoint())
+		s.Equal(storage.CollectionMethod_CORE_BPF, cluster.GetCollectionMethod())
+		if tc := cluster.GetTolerationsConfig(); s.NotNil(tc) {
+			s.False(tc.GetDisabled())
+		}
+		if dc := cluster.GetDynamicConfig(); s.NotNil(dc) {
+			s.True(dc.GetDisableAuditLogs())
+			if acc := dc.GetAdmissionControllerConfig(); s.NotNil(acc) {
+				s.False(acc.GetEnabled())
+				s.Equal(int32(defaultAdmissionControllerTimeout),
+					acc.GetTimeoutSeconds())
+			}
+		}
+	})
```

Comment:
Alternative to prevent nil pointer panics (Require will interrupt the test if the tested condition is not met):
```suggestion
		tc := cluster.GetTolerationsConfig()
		s.Require().NotNil(tc)
		s.False(tc.GetDisabled())

		dc := cluster.GetDynamicConfig()
		s.Require().NotNil(dc)
		s.True(dc.GetDisableAuditLogs())
		acc := dc.GetAdmissionControllerConfig()
		s.Require().NotNil(acc)
		s.False(acc.GetEnabled())
		s.Equal(int32(defaultAdmissionControllerTimeout), acc.GetTimeoutSeconds())
	})
```
Applies to other similar test code paths.